Assets/Scripts/UI/MainMenu/QueueMenu.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class QueueMenu : MonoBehaviour
+{
+    private void OnEnable()
+    {
+        SteamManager.Singleton.LobbyListUpdate += SetUpQueue;
+    }
+    private void OnDisable()
+    {
+        SteamManager.Singleton.LobbyListUpdate -= SetUpQueue;
+    }
+    private void SetUpQueue()
+    {
+        var viableLobbies = SteamManager.Singleton.Lobbies
+            .Where(lobby => lobby.gameMode == MatchRules.Singleton.Rules.GameMode)
+            .OrderBy(lobby => lobby.availableSlots);
+
+        if (viableLobbies.ToArray().Length > 1)
+        {
+            var enumerator = viableLobbies.GetEnumerator();
+            do
+                if (SteamManager.Singleton.RequestLobbyJoin(viableLobbies.First().id))
+                    return;
+            while (enumerator.MoveNext());
+        }
+
+        // No matching lobbies found, creating new
```

Comment:
I can debug it out, but creating an in game log with feedback for user goes into the issue-pile

Reply:
Okay we'll do it later ™️ 
I just thought it could be like "searching for open lobbies..." and then "no lobbies found, hosting new lobby" in a loading screen or smth